helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -502,15 +506,20 @@ private List<String> batchCustomInstanceStoppableCheck(String clusterId, List<St
       return instances;
     }
 
+    // Skip performing a custom check on any dead instance if the user set _skipCustomCheckIfInstanceNotAlive
+    // to true. This is because we do not attempt custom checks for instances that are already down.
+    List<String> liveInstanceIds = filterOutDeadInstancesIfNeeded(instances);
```

Comment:
Although more lines of code is needed, but do you think it is clearer to
List<String> instanceIdForAggregatedCustomCheck = _skipCustomChecksIfNoLiveness? filterOutDeadInstancesIfNeeded() : instances. 

It is a bit confusing here since `liveInstanceIds` may actually contains non live instances